src/main/java/bio/terra/workspace/service/workspace/flight/PullProjectFromPoolStep.java

```diff
@@ -30,13 +33,14 @@ public PullProjectFromPoolStep(
   public StepResult doStep(FlightContext flightContext) {
     try {
       String resourceId = flightContext.getWorkingMap().get(RBS_RESOURCE_ID, String.class);
+      logger.info("Preparing to query Buffer Service for resource with ID: " + resourceId);
       HandoutRequestBody body = new HandoutRequestBody();
       body.setHandoutRequestId(resourceId);
 
       ResourceInfo info = bufferService.handoutResource(body);
-      flightContext
-          .getWorkingMap()
-          .put(GOOGLE_PROJECT_ID, info.getCloudResourceUid().getGoogleProjectUid().getProjectId());
+      String projectId = info.getCloudResourceUid().getGoogleProjectUid().getProjectId();
+      logger.info("Buffer Service returned project with id: " + projectId);
+      flightContext.getWorkingMap().put(GOOGLE_PROJECT_ID, projectId);
       return StepResult.getStepResultSuccess();
     } catch (BufferServiceAPIException e) {
```

Comment:
should we examine the error code to determine if it is transient?

Reply:
Yes, we probably should. I will ask Yonghao. For the moment put a TODO as would like to get this in.